coala_quickstart/generation/SettingsClass.py

```diff
@@ -0,0 +1,256 @@
+from coala_quickstart.generation.Utilities import (
+    search_for_orig, get_all_args, get_default_args)
+
+
+def in_annot(func, key):
+    """
+    Checks if a setting name as key is present in function
+    annotations.
+    :param func:
+        Function object.
+    :param key:
+        The setting name as a string.
+    :return:
+        The value of type annotated or False.
+    """
+    return func.__annotations__[key] if key in func.__annotations__ else False
+
+
+def in_default_args(func, key):
+    """
+    Checks if a setting name as key is present in function
+    annotations.
+    :param func:
+        Function object.
+    :param key:
+        The setting name as a string.
+    :return:
+        True if key is present in args with default value else False.
+    """
+    return True if key in get_default_args(func) else False
+
+
+def in_all_args(func, key):
+    """
+    Checks if a setting name as key is present in function
+    annotations.
+    :param func:
+        Function object.
+    :param key:
+        The setting name as a string.
+    :return:
+        True if key is present in args of a function else False.
+    """
+    return True if key in get_all_args(func) else False
+
+
+class SettingTypes:
+
+    """
+    Categorizes the settings into Type bool and Type others
```

Comment:
So it's type 1 and the rest right? (acc. to cEP)

Reply:
Yes